tests/Tests.Reproduce/GitHubIssue281.cs

```diff
@@ -0,0 +1,195 @@
+*  not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+* 	http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using OpenSearch.OpenSearch.Xunit.XunitPlumbing;
+using OpenSearch.Net;
+using FluentAssertions;
+using OpenSearch.Client;
+using Xunit;
+
+namespace Tests.Reproduce
+{
+	public class GithubIssue281
+	{
+		public class SampleDomainObject
+		{
+			[JsonPropertyName("first_name")]
+			public string FirstName { get; set; }
+
+			[JsonPropertyName("last_name")]
+			public string LastName { get; set; }
+		}
+
+		[U]
+		public void GithubIssu281_MustNotWithTermQueryAndVerbatimEmptyValueShouldBeInRequestBody()
+		{
+			var connectionSettings = new ConnectionSettings(new InMemoryConnection()).DisableDirectStreaming();
+			var client = new OpenSearchClient(connectionSettings);
+
+			var action = () =>
+				client.Search<SampleDomainObject>(s => s
+					.Query(q => q
+						.Bool(b => b
+							.Must(m => m.Exists(e => e.Field("last_name")))
+							.MustNot(m => m.Term(t => t.Verbatim().Field("last_name.keyword").Value(string.Empty)))
+						)
+					)
+					.Index("index")
+					.Source(sfd => null)
+				);
+
+			var response = action.Should().NotThrow().Subject;
+
+			var json = Encoding.UTF8.GetString(response.ApiCall.RequestBodyInBytes);
+			json.Should()
+				.Be(
+					@"{""query"":{""bool"":{""must"":[{""exists"":{""field"":""last_name""}}],""must_not"":[{""term"":{""last_name.keyword"":{""value"":""""}}}]}}}");
+		}
+
+		[U]
+		public void GithubIssue281_MustNotTermQueryAndVerbatimEmptyValueShouldBeRegisteredAsNonNull()
+		{
+			Func<SearchDescriptor<SampleDomainObject>, ISearchRequest> selector = s => s
+				.Query(q => q
+					.Bool(b => b
+						.Must(m => m.Exists(e => e.Field("last_name")))
+						.MustNot(m => m.Term(t => t.Verbatim().Field("last_name.keyword").Value(string.Empty)))
+					)
+				)
+				.Index("index")
+				.Source(sfd => null);
+
+			var searchRequest = selector.Invoke(new SearchDescriptor<SampleDomainObject>());
+			var query = searchRequest.Query as IQueryContainer;
+
+			// this is fine
+			query.Bool.Must.Should().NotBeEmpty();
+			query.Bool.Must.First().Should().NotBeNull("Must");
+
+			// this too...
+			query.Bool.MustNot.Should().NotBeEmpty();
+			// ... and this passes so long as `.Verbatim()` is used in the `TermQuery`
+			query.Bool.MustNot.First().Should().NotBeNull("MustNot");
+		}
+
+		[U]
+		public void GithubIssue281_MustNotTermQueryAndNonVerbatimEmptyValueShouldBeRegisteredAsNonNull()
```

Comment:
Looks like this should be `NonEmptyValue`:
```suggestion
		public void GithubIssue281_MustNotTermQueryAndNonVerbatimNonEmptyValueShouldBeRegisteredAsNonNull()
```

Reply:
good catch! fixed.